test/katex-spec.js

```diff
@@ -1484,6 +1484,35 @@ describe("A font parser", function() {
     });
 });
 
+describe("A comment parser", function() {
+    it("should parse comments at the end of a line", () => {
+        expect("a^2 + b^2 = c^2 % Pythagoras' Theorem\n").toParse();
+    });
+
+    it("should parse comments at the start of a line", () => {
+        expect("% comment\n").toParse();
+    });
+
+    it("should parse multiple lines of comments in a row", () => {
+        expect("% comment 1\n% comment 2\n").toParse();
+    });
+
+    it("should not parse a comment that isn't followed by a newline", () => {
+        expect("x%y").toNotParse();
```

Comment:
This seems a little strange to me. Maybe we could support comments that go to the end of the input string as well as ones that end with a newline? I feel like this will catch people up otherwise.

Reply:
There was recent discussion in #20 about this.  If we allow comments without the newline then some KaTeX strings will break when copied over to a LaTeX document.